scheduler/stack.go

```diff
@@ -62,16 +65,21 @@ func NewGenericStack(batch bool, ctx Context) *GenericStack {
 	s.source = NewRandomIterator(ctx, nil)
 
 	// Attach the job constraints. The job is filled in later.
-	s.jobConstraint = NewConstraintIterator(ctx, s.source, nil)
+	s.jobConstraint = NewConstraintChecker(ctx, nil)
 
 	// Filter on task group drivers first as they are faster
-	s.taskGroupDrivers = NewDriverIterator(ctx, s.jobConstraint, nil)
+	s.taskGroupDrivers = NewDriverChecker(ctx, nil)
 
 	// Filter on task group constraints second
-	s.taskGroupConstraint = NewConstraintIterator(ctx, s.taskGroupDrivers, nil)
+	s.taskGroupConstraint = NewConstraintChecker(ctx, nil)
+
+	// Create the feasibility wrapper.
```

Comment:
Add an extra comment as to why this wrapper is used and what kinds of things should be part of it for future context.